src/test/scala/freeeventsourcing/AggregateEventSelectorTests.scala

```diff
@@ -0,0 +1,102 @@
+package freeeventsourcing
+
+import org.scalatest.{ FlatSpec, Matchers }
+import shapeless.{ ::, HNil }
+import freeeventsourcing.EventSelector.WithEventType
+import freeeventsourcing.EventSelector.ops._
+import freeeventsourcing.ProcessActionTests.{ OtherEvent, TestAggregate }
+import freeeventsourcing.accountprocessing.Account.Event.{ Closed, Opened }
+import freeeventsourcing.accountprocessing.Transaction.Event.Created
+import freeeventsourcing.accountprocessing._
+import freeeventsourcing.utils.CompositeName
+
+class AggregateEventSelectorTests extends FlatSpec with Matchers {
+  val router = AggregateEventSelector.Router.forAggregateType(Account)
+
+  val openedEvent = AggregateEvent[Account.type, Opened](Account, Account.Id(1), Opened("Mario"))
+
+  "AggregateEventSelector.Router " should " produce the same topic as the matching AggregateEventSelector" in {
+    val toIndex = router.apply(openedEvent)
+    toIndex.size shouldBe 1
+    val indexed = toIndex.head
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/msiegenthaler/free-event-sourcing/file/2651771996/issues/source?bid=3298199&fileBranchId=3311368#l21)